tests/test_run_reference_vmm.py

```diff
@@ -225,16 +225,37 @@ def expect_vcpus(vmm_process, expected_vcpus):
     # dependent on it, so it's just fine to call it again, less than ideal, but not
     # wrong.
     setup_stdout_nonblocking(vmm_process)
+    giveup_after = timeout
+    then = time.time()
+    cnt=0
+    while True:
```

Comment:
I don't understand why this change is needed. Can you please explain it? Also, it should be part of a separate commit.

Reply:
If we need something more complex than just a `vmm_process.stdin.write(b'cmd')` then we should have this functionality in a separate function, and then call it in all the other functions as well. It's strange to have different ways of sending commands to the VMM depending on the test function.